lib/phoenix/socket/transport.ex

```diff
@@ -94,30 +94,6 @@ defmodule Phoenix.Socket.Transport do
 
   @type state :: term()
 
-  @doc """
-  Returns a child specification for socket management.
-
-  This is invoked only once per socket regardless of
-  the number of transports and should be responsible
-  for setting up any process structure used exclusively
-  by the socket regardless of transports.
-
-  Each socket connection is started by the transport
-  and the process that controls the socket likely
-  belongs to the transport. However, some sockets spawn
-  new processes, such as `Phoenix.Socket` which spawns
-  channels, and this gives the ability to start a
-  supervision tree associated to the socket.
-
-  It receives the socket options from the endpoint,
-  for example:
-
-      socket "/my_app", MyApp.Socket, shutdown: 5000
-
-  means `child_spec([shutdown: 5000])` will be invoked.
-  """
-  @callback child_spec(keyword) :: :supervisor.child_spec()
```

Comment:
Why was this moved to Phoenix.Socket? I don't think it is actually part of Phoenix.Socket, because the user does not need to define a `child_spec` for `Phoenix.Socket`. :)

Reply:
(Note that we've [talked about this before](https://github.com/phoenixframework/phoenix/pull/4973#issuecomment-1261041615), but I suppose never came to a conclusion on it)

The actual concrete implementation of this callback hasn't changed - it's still `use`d into the Socket at https://github.com/mtrudel/phoenix/blob/af00565585bdbc5644f9ed0a1077d52d87ac3f0f/lib/phoenix/socket.ex#L309-L311. The only thing that changed here is which behaviour this is part of. The scope of this callback based on its docs makes this seem like a better home for it. Not strictly blocking, though.

Reply:
It should be part of `Phoenix.Socket.Transport` because someone can implement `Phoenix.Socket.Transport` so they bypass the whole channel machinery and I believe that, without this callback, it won't work.

Reply:
Updated!